### PR TITLE
Feature/11 setup frontend routes n states

### PR DESCRIPTION
**Pull request by wayangalihpratama**
### TODO / DONE

---

- #11 
- [x] Setup folder structure
- [x] Setup route
- [x] Setup context and recuder
- [x] Create initial page design

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1207380167122107

### CODE REVIEW

File: backend/routes/user_routes.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+from utils.jwt_handler import create_jwt_token
+
+router = APIRouter()
+webdomain = environ.get("WEBDOMAIN")
+
+
+@router.post("/login")
+async def send_login_link(
+    phone_number: int, session: Session = Depends(get_session)
+):
```
Comment (kjkoster) [inline backend/routes/user_routes.py:18]: Before using the user input, validate it to ensure it is a clean, globalised E.164 number. Regex: `r'^\+[1-9]\d{1,14}$'`.

File: backend/routes/user_routes.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    user = session.exec(
+        select(User).where(User.phone_number == phone_number)
+    ).first()
+    if not user:
+        raise HTTPException(status_code=404, detail="User not found")
```
Comment (kjkoster) [inline backend/routes/user_routes.py:23]: As a first implementation, we can create the user here and send them a login link. That way we have a trivial onboarding. We will have to keep an eye on the registrations, to check that no-one will try anything funny, but we won't need an admin interface until much later into the project. Until then, PGAdmin will serve as our admin interface.

File: frontend/next.config.mjs [hunk 3]
```diff
@@ -5,7 +5,7 @@ const env = dotenv.config();
 const backend_port = env.parsed.BACKEND_PORT;
 
 const nextConfig = {
-  output: "export",
+  // output: "export",
```
Comment (kjkoster) [inline frontend/next.config.mjs:8]: Please don't commit commented code.

**Review by kjkoster — CHANGES_REQUESTED** [review]
I cannot review JavaScript much, sorry. We can do that interactively if you like.

File: backend/routes/user_routes.py (hunk 2 above)
Comment (dedenbangkit) [reply backend/routes/user_routes.py]: solved by database seeder CLI

**Review by dedenbangkit — APPROVED** [review]

File: backend/routes/user_routes.py (hunk 1 above)
Comment (dedenbangkit) [reply backend/routes/user_routes.py:30]: https://github.com/akvo/rag-doll/pull/12/commits/41b5b0c8e43281fbcce4f64961b575b9549cb9f4

**Review by kjkoster — APPROVED** [review]

**Pull request merged**